Test that `network` is `Testnet` in HttpBridgeSpec

Anviking opened the pull request:
# Overview

- [x] Test that `network` is `Testnet` in integration/HttpBridgeSpec

# Comments
- When someone runs the tests with NETWORK=Mainnet/Staging/Local we now
make it clearer that this is unsupported.
- The specific failure reasons — from the other tests — are still shown:

```
test/integration/Cardano/Wallet/Network/HttpBridgeSpec.hs:62:9:
      1) Cardano.Wallet.Network.HttpBridge The NETWORK env var must be testnet for these tests
           expected: Testnet
            but got: Mainnet

      To rerun use: --match "/Cardano.Wallet.Network.HttpBridge/The NETWORK env var must be testnet for these tests/"

      test/integration/Cardano/Wallet/Network/HttpBridgeSpec.hs:68:13:
      2) Cardano.Wallet.Network.HttpBridge, Happy paths, get from packed epochs
           expected: Right (Hash {getHash = "7Z\213\204\SUB\134l\149\&8\191ZO\\0q]\ESCB\CAN\254\f[\RS\142\SOH\192K\250^\168\188m"})
            but got: Right (Hash {getHash = "\221_-@n\166\198+n7\SI\137\204\&9\216\SO@\NAK-[\NAK7\196\DC3\135\&5\158\"\r\223_\146"})

      To rerun use: --match "/Cardano.Wallet.Network.HttpBridge/Happy paths/get from packed epochs/"

      test/integration/Cardano/Wallet/Network/HttpBridgeSpec.hs:76:13:
      3) Cardano.Wallet.Network.HttpBridge, Happy paths, get from packet epochs and filter by start slot
           expected: Right 7600
            but got: Right 7599

      To rerun use: --match "/Cardano.Wallet.Network.HttpBridge/Happy paths/get from packet epochs and filter by start slot/"

      test/integration/Cardano/Wallet/Network/HttpBridgeSpec.hs:139:13:
      4) Cardano.Wallet.Network.HttpBridge, Submitting signed transactions, old tx fails
           expected: Left (ErrPostTxProtocolFailure "Failed to send to peers: Blockchain protocol error")
            but got: Left (ErrPostTxBadRequest "Transaction failed verification: invalid transaction witness")

      To rerun use: --match "/Cardano.Wallet.Network.HttpBridge/Submitting signed transactions/old tx fails/"

    Randomized with seed 1462693906

    Finished in 18.6094 seconds
    73 examples, 4 failures
```
<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

denisshevchenko commented on src/Cardano/Wallet/Kernel/DB/EosHdWallet/Create.hs:
```diff
@@ -73,6 +84,21 @@ createEosHdAccount eosHdAccount = do
+        -- We don't have 'AutoIncrementKey' for 'EosHdAccount', but since we use
+        -- the same type hierarchy as for 'HdAddress', we have to provide fake
+        -- 'AutoIncrementKey' value.
+        let fakeAutoIncrementKey = IxSet.AutoIncrementKey 0
```
> What is this AutoIncrementKey used for?

I have no idea. :-) But since we use the _same_ hierarchy - it requires `AutoIncrementKey`.

KtorZ replied:
Then, could you get an idea :smile:?

denisshevchenko commented:
Closed because of inappropriate changes (new PRs will be opened instead).

Anviking committed:
Test that `network` is `Testnet` in HttpBridgeSpec

When someone runs the tests with NETWORK=Mainnet/Staging/Local we now
make it clearer that this is unsupported.

The specific failure reasons — from the other tests — are still shown.
